src/server_state.cpp

```diff
@@ -565,29 +565,45 @@ wsrep::server_state::causal_read(int timeout) const
     return provider_->causal_read(timeout);
 }
 
-void wsrep::server_state::on_connect(const wsrep::gtid& gtid)
+void wsrep::server_state::on_connect(const wsrep::view& view)
 {
+    assert(id_.is_undefined());
+    id_ = view.members()[view.own_index()].id();
```

Comment:
It would be good to have some input validation for `view.own_index()`. Assert/throw if it is negative or `>= view.members().size()`.